### ISSUE TITLE
Cognito Amplify Auth.confirmSignIn fails

### ISSUE THREAD

**Issue by shouvik44** [aws/aws-amplify#651]
Do you want to request a feature or report a bug?
- A bug

What is the current behavior?
- I've manually configured Auth/Cognito in 2 apps, one is a mobile app in React-Native and one is a web app in React.
- I've required MFA with an SMS
- I can login and confirm signin just fine in the React-Native mobile app
- I can login - but can't confirm sign in in the web app
- The React/Redux function to confirm signin is the same in both places. The only difference is, in the Amplify.Configure step, I change the userPoolWebClientId to the app's client id.

When I successfully login and then confirm with the MFA code, AWS returns a "signin failed" error.  Looking at the [Auth/Cognito code](https://github.com/aws/aws-amplify/blob/968dde8/packages/aws-amplify/src/Auth/Auth.ts#L81), that means there's some error in the onSuccess branch here

/**
     * Send MFA code to confirm sign in
     * @param {Object} user - The CognitoUser object
     * @param {String} code - The confirmation code
     */
    public confirmSignIn(user: any, code: string, mfaType: string | null): Promise<any> {
        if (!code) { return Promise.reject('Code cannot be empty'); }

        const that = this;
        return new Promise((resolve, reject) => {
            user.sendMFACode(
                code, {
                    onSuccess: (session) => {
                        logger.debug(session);
                        that._setCredentialsFromSession(session).then((cred) => {
                            that.user = user;
                            dispatchAuthEvent('signIn', user);
                            resolve(user);
                        }).catch(e => {
                            logger.debug('cannot get cognito credentials');
                            reject('signin failed');
                        });
                    },
                    onFailure: (err) => {
                        logger.debug('confirm signIn failure', err);
                        reject(err);
                    }
                }, 
                mfaType);
        });
    }

**Comment by richardzcode** [comment]
Hey @shouvik44 could you turn LOG_LEVEL to 'DEBUG'? So we can get detailed logs.

`Amplify.Logger.LOG_LEVEL = 'DEBUG'`

**Comment by richardzcode** [comment]
We have another issue #646 may be same. From network and logs looks like `identityPoolId` config is the potential cause.

@shouvik44 is it okay for you to share your `identityPoolId` config? If not, could you confirm if it follows pattern of ` [\w-]+:[0-9a-f-]+`?

Basically that means something like: us-east-1:b0ee1234-5ae6-7c89-0123-45dd67890c1d, notice latter part must not have letters outside of a - f.

**Comment by shouvik44** [comment]
Hey @richardzcode - I ran the detailed logs and am seeing these issues. 
1) AuthClass - gettingSessionSuccess.
2) AuthClass - set credentials from session
Post 1: There's a Post to the cognito-identity server that's returning a 400 error.
3) AuthClass: "Failed to load credentials: CognitoIdentityCredentials" with an object the expired is true, expireTime is null, and accessKeyId is undefined.
4) AuthClass: sets credentials for guest.
Post 2: Another Post to cognito with a 400 error.
5) AuthClass: failed to load credentials
6) AnalysticsClass: ensure credentials error (3x)
7) AuthClass - signin MFA required
8) AuthClass - CognitoUserSession
9) AuthClass - set credentials from session
Post 3 to cognito with 400 error
10) Auth class - Failed to load credentials
11) Auth class - cannot get cognito credentials
12) signin failed

**Comment by shouvik44** [comment]
@richardzcode The identityPoolId is us-east-1:23b582de-690e-43a0-b9c5-fff3deac112d

Think that passes the regex test you gave.

**Comment by richardzcode** [comment]
it is same as #646 . confirm MFA succeeded but load credentials failed. The server side complain about 'identityPoolId' is invalid.

Could you run `npm install aws-amplify@latest`? Just to make sure.

**Comment by richardzcode** [comment]
In logs,

right after '[DEBUG] ..:..:... AuthClass - set credentials from session',
and before '[DEBUG] ..:..:... AuthClass - getting session failed "Failed to load credentials",
there is a CognitoIdentityCredentials object.

Could you expend it, and check 'params' -> 'identityPoolId' value of it?

**Comment by shouvik44** [comment]
I expanded CognitoIdentityCredentials and params.IdentityPoolId is what I
expect -  us-east-1:23b582de-690e-43a0-b9c5-fff3deac112d.

However, after the npm install to the latest aws-amplify, I'm seeing more
details in the error. After AuthClass - Failed to Load Credentials, am
seeing

"AuthClass - getting session failed: Error: Token is not from a supporter
provided of this identity pool."

Here's the updated Debug messages

   - {[DEBUG] 56:34.67 AuthClass - getting session success:
   CognitoUserSession}[DEBUG] 56:34.67 AuthClass - getting session success:
   CognitoUserSessionaccessToken: CognitoAccessToken {jwtToken:
   "eyJraWQiOiJWQVwvOWF3dkd4OStmblNqUWV6ZmZzc25rU0pMdG…B2GcUqcqvSfACVg9cCv8ul2qzfmDmKH2GVeljY_N6JjmNiw5A",
   payload: {…}}clockDrift: 0idToken: CognitoIdToken {jwtToken:
   "eyJraWQiOiJCSlRwN05tc0I5eDlvUzNRUHgxajA4b3hkYVNvYl…yCqL4mtEf_VYYYVgdxcccQ2ZyujZGHztJbU2awYpmxf6dFdDg",
   payload: {…}}refreshToken: CognitoRefreshToken {token:
   "eyJjdHkiOiJKV1QiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiUl…GaaBPRzPnyf8_L-ns6AQQgAYzI.zaf2PYibwrClqzrbkssKPg"}__proto__:
   Object__proto__: Object
   - ConsoleLogger.js:84 [DEBUG] 56:34.68 AuthClass - set credentials from
   session
   - ConsoleLogger.js:100 {[DEBUG] 56:34.971 AuthClass - Failed to load
   credentials: CognitoIdentityCredentials}[DEBUG] 56:34.971 AuthClass -
   Failed to load credentials: CognitoIdentityCredentialsaccessKeyId:
   undefinedcognito: features.constructor {config: Config, isGlobalEndpoint:
   false, endpoint: Endpoint, _clientId: 1}data: nullexpireTime: nullexpired:
   trueparams: IdentityPoolId:
   "us-east-1:23b582de-690e-43a0-b9c5-fff3deac112d"Logins: {
   cognito-idp.us-east-1.amazonaws.com/us-east-1_gxFMkeRzN:
   "eyJraWQiOiJCSlRwN05tc0I5eDlvUzNRUHgxajA4b3hkYVNvYl…yCqL4mtEf_VYYYVgdxcccQ2ZyujZGHztJbU2awYpmxf6dFdDg"}RoleSessionName:
   "web-identity"__proto__: ObjectsessionToken: undefinedsts:
   features.constructor {config: Config, isGlobalEndpoint: true, endpoint:
   Endpoint, _clientId: 2}webIdentityCredentials: WebIdentityCredentials
   {expired: true, expireTime: null, accessKeyId: undefined, sessionToken:
   undefined, params: {…}, …}_clientConfig: {region: "us-east-1"}_identityId:
   nullidentityId: (...)secretAccessKey: undefinedget identityId: ƒ ()set
   identityId: ƒ (identityId)__proto__: Credentials__proto__: Object
   - ConsoleLogger.js:100 {[DEBUG] 56:34.972 AuthClass - getting session
   failed: Error: Token is not from a supported provider of this identity pool.
   -     at Request.extractError (ht…}[DEBUG] 56:34.972 AuthClass - getting
   session failed: Error: Token is not from a supported provider of this
   identity pool.
   -     at Request.extractError (
   http://localhost:3000/static/js/bundle.js:74329:27)
   -     at Request.callListeners (
   http://localhost:3000/static/js/bundle.js:77321:20)
   -     at Request.emit (http://localhost:3000/static/js/bundle.js:77293:10
   )
   -     at Request.emit (http://localhost:3000/static/js/bundle.js:75930:14
   )
   -     at Request.transition (
   http://localhost:3000/static/js/bundle.js:75269:10)
   -     at AcceptorStateMachine.runTo (
   http://localhost:3000/static/js/bundle.js:80785:12)
   -     at http://localhost:3000/static/js/bundle.js:80797:10
   -     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75285:9)
   -     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75932:12)
   -     at Request.callListeners (
   http://localhost:3000/static/js/bundle.js:77331:18)
   -     at Request.emit (http://localhost:3000/static/js/bundle.js:77293:10
   )
   -     at Request.emit (http://localhost:3000/static/js/bundle.js:75930:14
   )
   -     at Request.transition (
   http://localhost:3000/static/js/bundle.js:75269:10)
   -     at AcceptorStateMachine.runTo (
   http://localhost:3000/static/js/bundle.js:80785:12)
   -     at http://localhost:3000/static/js/bundle.js:80797:10
   -     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75285:9)
   -     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75932:12)
   -     at Request.callListeners (
   http://localhost:3000/static/js/bundle.js:77331:18)
   -     at callNextListener (
   http://localhost:3000/static/js/bundle.js:77311:12)
   -     at EventEmitter.onEnd (
   http://localhost:3000/static/js/bundle.js:72411:13)
   -     at EventEmitter../node_modules/events/events.js.EventEmitter.emit (
   http://localhost:3000/static/js/bundle.js:115858:17)
   -     at features.constructor.finishRequest (
   http://localhost:3000/static/js/bundle.js:73093:13)
   -     at XMLHttpRequest.<anonymous> (
   http://localhost:3000/static/js/bundle.js:73008:14)code:
   "NotAuthorizedException"message: "Token is not from a supported provider of
   this identity pool."requestId:
   "016014ba-3f44-11e8-bb9c-b9942dd2cfad"retryDelay:
   34.50353615853725retryable: falsestatusCode: 400time: Fri Apr 13 2018
   10:56:34 GMT-0700 (PDT) {}name: "NotAuthorizedException"stack:
   "NotAuthorizedException: Token is not from a supported provider of this
   identity pool.↵    at Request.extractError (
   http://localhost:3000/static/js/bundle.js:74329:27)↵    at
   Request.callListeners (
   http://localhost:3000/static/js/bundle.js:77321:20)↵    at Request.emit (
   http://localhost:3000/static/js/bundle.js:77293:10)↵    at Request.emit (
   http://localhost:3000/static/js/bundle.js:75930:14)↵    at
   Request.transition (http://localhost:3000/static/js/bundle.js:75269:10)↵
     at AcceptorStateMachine.runTo (
   http://localhost:3000/static/js/bundle.js:80785:12)↵    at
   http://localhost:3000/static/js/bundle.js:80797:10↵    at
   Request.<anonymous> (http://localhost:3000/static/js/bundle.js:75285:9)↵
     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75932:12)↵    at
   Request.callListeners (
   http://localhost:3000/static/js/bundle.js:77331:18)↵    at Request.emit (
   http://localhost:3000/static/js/bundle.js:77293:10)↵    at Request.emit (
   http://localhost:3000/static/js/bundle.js:75930:14)↵    at
   Request.transition (http://localhost:3000/static/js/bundle.js:75269:10)↵
     at AcceptorStateMachine.runTo (
   http://localhost:3000/static/js/bundle.js:80785:12)↵    at
   http://localhost:3000/static/js/bundle.js:80797:10↵    at
   Request.<anonymous> (http://localhost:3000/static/js/bundle.js:75285:9)↵
     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75932:12)↵    at
   Request.callListeners (
   http://localhost:3000/static/js/bundle.js:77331:18)↵    at
   callNextListener (http://localhost:3000/static/js/bundle.js:77311:12)↵
     at EventEmitter.onEnd (
   http://localhost:3000/static/js/bundle.js:72411:13)↵    at
   EventEmitter../node_modules/events/events.js.EventEmitter.emit (
   http://localhost:3000/static/js/bundle.js:115858:17)↵    at
   features.constructor.finishRequest (
   http://localhost:3000/static/js/bundle.js:73093:13)↵    at
   XMLHttpRequest.<anonymous> (
   http://localhost:3000/static/js/bundle.js:73008:14)"__proto__:
   Object__proto__: Object
   - ConsoleLogger.js:84 [DEBUG] 56:34.973 AuthClass - setting credentials
   for guest
   - cognito-identity.us-east-1.amazonaws.com/:1 POST
   https://cognito-identity.us-east-1.amazonaws.com/ 400 ()
   - ConsoleLogger.js:100 {[DEBUG] 56:35.156 AuthClass - Failed to load
   credentials: CognitoIdentityCredentials}[DEBUG] 56:35.156 AuthClass -
   Failed to load credentials: CognitoIdentityCredentials {expired: true,
   expireTime: null, accessKeyId: undefined, sessionToken: undefined, params:
   {…}, …}accessKeyId: undefinedcognito: features.constructor {config: Config,
   isGlobalEndpoint: false, endpoint: Endpoint, _clientId: 3}data:
   nullexpireTime: nullexpired: trueparams: {IdentityPoolId:
   "us-east-1:23b582de-690e-43a0-b9c5-fff3deac112d", RoleSessionName:
   "web-identity"}sessionToken: undefinedsts: features.constructor {config:
   Config, isGlobalEndpoint: true, endpoint: Endpoint, _clientId:
   4}webIdentityCredentials: WebIdentityCredentials {expired: true,
   expireTime: null, accessKeyId: undefined, sessionToken: undefined, params:
   {…}, …}_clientConfig: {region: "us-east-1"}_identityId: nullidentityId:
   (...)secretAccessKey: undefinedget identityId: ƒ ()set identityId: ƒ
   (identityId)__proto__: Credentials__proto__: Object
   - ConsoleLogger.js:100 {[DEBUG] 56:35.157 AnalyticsClass - ensure
   credentials error: Error: Unauthenticated access is not supported for this
   identity pool.
   -     at Request.extractError (…}[DEBUG] 56:35.157 AnalyticsClass -
   ensure credentials error: Error: Unauthenticated access is not supported
   for this identity pool.
   -     at Request.extractError (
   http://localhost:3000/static/js/bundle.js:74329:27)
   -     at Request.callListeners (
   http://localhost:3000/static/js/bundle.js:77321:20)
   -     at Request.emit (http://localhost:3000/static/js/bundle.js:77293:10
   )
   -     at Request.emit (http://localhost:3000/static/js/bundle.js:75930:14
   )
   -     at Request.transition (
   http://localhost:3000/static/js/bundle.js:75269:10)
   -     at AcceptorStateMachine.runTo (
   http://localhost:3000/static/js/bundle.js:80785:12)
   -     at http://localhost:3000/static/js/bundle.js:80797:10
   -     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75285:9)
   -     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75932:12)
   -     at Request.callListeners (
   http://localhost:3000/static/js/bundle.js:77331:18)
   -     at Request.emit (http://localhost:3000/static/js/bundle.js:77293:10
   )
   -     at Request.emit (http://localhost:3000/static/js/bundle.js:75930:14
   )
   -     at Request.transition (
   http://localhost:3000/static/js/bundle.js:75269:10)
   -     at AcceptorStateMachine.runTo (
   http://localhost:3000/static/js/bundle.js:80785:12)
   -     at http://localhost:3000/static/js/bundle.js:80797:10
   -     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75285:9)
   -     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75932:12)
   -     at Request.callListeners (
   http://localhost:3000/static/js/bundle.js:77331:18)
   -     at callNextListener (
   http://localhost:3000/static/js/bundle.js:77311:12)
   -     at EventEmitter.onEnd (
   http://localhost:3000/static/js/bundle.js:72411:13)
   -     at EventEmitter../node_modules/events/events.js.EventEmitter.emit (
   http://localhost:3000/static/js/bundle.js:115858:17)
   -     at features.constructor.finishRequest (
   http://localhost:3000/static/js/bundle.js:73093:13)
   -     at XMLHttpRequest.<anonymous> (
   http://localhost:3000/static/js/bundle.js:73008:14)code:
   "NotAuthorizedException"message: "Unauthenticated access is not supported
   for this identity pool."requestId:
   "018aa73a-3f44-11e8-878f-2d7daf662dc3"retryDelay:
   45.57173023809733retryable: falsestatusCode: 400time: Fri Apr 13 2018
   10:56:35 GMT-0700 (PDT) {}name: "NotAuthorizedException"stack:
   "NotAuthorizedException: Unauthenticated access is not supported for this
   identity pool.↵    at Request.extractError (
   http://localhost:3000/static/js/bundle.js:74329:27)↵    at
   Request.callListeners (
   http://localhost:3000/static/js/bundle.js:77321:20)↵    at Request.emit (
   http://localhost:3000/static/js/bundle.js:77293:10)↵    at Request.emit (
   http://localhost:3000/static/js/bundle.js:75930:14)↵    at
   Request.transition (http://localhost:3000/static/js/bundle.js:75269:10)↵
     at AcceptorStateMachine.runTo (
   http://localhost:3000/static/js/bundle.js:80785:12)↵    at
   http://localhost:3000/static/js/bundle.js:80797:10↵    at
   Request.<anonymous> (http://localhost:3000/static/js/bundle.js:75285:9)↵
     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75932:12)↵    at
   Request.callListeners (
   http://localhost:3000/static/js/bundle.js:77331:18)↵    at Request.emit (
   http://localhost:3000/static/js/bundle.js:77293:10)↵    at Request.emit (
   http://localhost:3000/static/js/bundle.js:75930:14)↵    at
   Request.transition (http://localhost:3000/static/js/bundle.js:75269:10)↵
     at AcceptorStateMachine.runTo (
   http://localhost:3000/static/js/bundle.js:80785:12)↵    at
   http://localhost:3000/static/js/bundle.js:80797:10↵    at
   Request.<anonymous> (http://localhost:3000/static/js/bundle.js:75285:9)↵
     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75932:12)↵    at
   Request.callListeners (
   http://localhost:3000/static/js/bundle.js:77331:18)↵    at
   callNextListener (http://localhost:3000/static/js/bundle.js:77311:12)↵
     at EventEmitter.onEnd (
   http://localhost:3000/static/js/bundle.js:72411:13)↵    at
   EventEmitter../node_modules/events/events.js.EventEmitter.emit (
   http://localhost:3000/static/js/bundle.js:115858:17)↵    at
   features.constructor.finishRequest (
   http://localhost:3000/static/js/bundle.js:73093:13)↵    at
   XMLHttpRequest.<anonymous> (
   http://localhost:3000/static/js/bundle.js:73008:14)"__proto__:
   Object__proto__: Object
   - ConsoleLogger.js:100 {[DEBUG] 56:35.158 AnalyticsClass - ensure
   credentials error: Error: Unauthenticated access is not supported for this
   identity pool.
   -     at Request.extractError (…}[DEBUG] 56:35.158 AnalyticsClass -
   ensure credentials error: Error: Unauthenticated access is not supported
   for this identity pool.
   -     at Request.extractError (
   http://localhost:3000/static/js/bundle.js:74329:27)
   -     at Request.callListeners (
   http://localhost:3000/static/js/bundle.js:77321:20)
   -     at Request.emit (http://localhost:3000/static/js/bundle.js:77293:10
   )
   -     at Request.emit (http://localhost:3000/static/js/bundle.js:75930:14
   )
   -     at Request.transition (
   http://localhost:3000/static/js/bundle.js:75269:10)
   -     at AcceptorStateMachine.runTo (
   http://localhost:3000/static/js/bundle.js:80785:12)
   -     at http://localhost:3000/static/js/bundle.js:80797:10
   -     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75285:9)
   -     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75932:12)
   -     at Request.callListeners (
   http://localhost:3000/static/js/bundle.js:77331:18)
   -     at Request.emit (http://localhost:3000/static/js/bundle.js:77293:10
   )
   -     at Request.emit (http://localhost:3000/static/js/bundle.js:75930:14
   )
   -     at Request.transition (
   http://localhost:3000/static/js/bundle.js:75269:10)
   -     at AcceptorStateMachine.runTo (
   http://localhost:3000/static/js/bundle.js:80785:12)
   -     at http://localhost:3000/static/js/bundle.js:80797:10
   -     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75285:9)
   -     at Request.<anonymous> (
   http://localhost:3000/static/js/bundle.js:75932:12)
   -     at Request.callListeners (
   http://localhost:3000/static/js/bundle.js:77331:18)
   -     at callNextListener (
   http://localhost:3000/static/js/bundle.js:77311:12)
   -     at EventEmitter.onEnd (
   http://localhost:3000/static/js/bundle.js:72411:13)
   -     at EventEmitter../node_modules/events/events.js.EventEmitter.emit (
   http://localhost:3000/static/js/bundle.js:115858:17)
   -     at features.constructor.finishRequest (
   http://localhost:3000/static/js/bundle.js:73093:13)
   -     at XMLHttpRequest.<anonymous> (
   http://localhost:3000/static/js/bundle.js:73008:14)__proto__: Object
   - ConsoleLogger.js:84 [DEBUG] 56:56.777 AuthClass - signIn MFA required
   - ConsoleLogger.js:90 {[DEBUG] 57:25.476 AuthClass:
   CognitoUserSession}[DEBUG] 57:25.476 AuthClass:
   CognitoUserSessionaccessToken: CognitoAccessToken {jwtToken:
   "eyJraWQiOiJWQVwvOWF3dkd4OStmblNqUWV6ZmZzc25rU0pMdG…keu2Elvhtpv0LYMwQ9qbwxFUXrlDts_23XzU1yrBliKY7Nmag",
   payload: {…}}clockDrift: 0idToken: CognitoIdToken {jwtToken:
   "eyJraWQiOiJCSlRwN05tc0I5eDlvUzNRUHgxajA4b3hkYVNvYl…fxswGz-diDCbQ5lvilht3Oo__XNTg3MQvD7SVYz6l9kJOaq5Q",
   payload: {…}}refreshToken: CognitoRefreshToken {token:
   "eyJjdHkiOiJKV1QiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiUl…1ShZjSg6j_K0Mjf72X5GyOVHFc.vqlfva0cBKPttyjTn8McSw"}__proto__:
   Object__proto__: Object
   - ConsoleLogger.js:84 [DEBUG] 57:25.477 AuthClass - set credentials from
   session
   - cognito-identity.us-east-1.amazonaws.com/:1 POST
   https://cognito-identity.us-east-1.amazonaws.com/ 400 ()
   - ConsoleLogger.js:100 {[DEBUG] 57:25.779 AuthClass - Failed to load
   credentials: CognitoIdentityCredentials}[DEBUG] 57:25.779 AuthClass -
   Failed to load credentials: CognitoIdentityCredentialsaccessKeyId:
   undefinedcognito: features.constructor {config: Config, isGlobalEndpoint:
   false, endpoint: Endpoint, _clientId: 5}config: Config {credentials: null,
   credentialProvider: null, region: "us-east-1", logger: null, apiVersions:
   {…}, …}endpoint: Endpoint {protocol: "https:", host: "
   cognito-identity.us-east-1.amazonaws.com", port: 443, hostname: "
   cognito-identity.us-east-1.amazonaws.com", pathname: "/",
   …}isGlobalEndpoint: false_clientId: 5_originalConfig: (...)get
   _originalConfig: ƒ ()__proto__: features.constructordata: nullexpireTime:
   nullexpired: trueparams: {IdentityPoolId:
   "us-east-1:23b582de-690e-43a0-b9c5-fff3deac112d", Logins: {…},
   RoleSessionName: "web-identity"}IdentityPoolId:
   "us-east-1:23b582de-690e-43a0-b9c5-fff3deac112d"Logins:
   cognito-idp.us-east-1.amazonaws.com/us-east-1_gxFMkeRzN:
   "eyJraWQiOiJCSlRwN05tc0I5eDlvUzNRUHgxajA4b3hkYVNvYlY5WjJMOFBDRTlcL2VzTT0iLCJhbGciOiJSUzI1NiJ9.eyJzdWIiOiJjNTJhZjlmZS1jM2U5LTRmYjAtOTFlZS0xNGJkNzdjNGIyMzEiLCJlbWFpbF92ZXJpZmllZCI6ZmFsc2UsImlzcyI6Imh0dHBzOlwvXC9jb2duaXRvLWlkcC51cy1lYXN0LTEuYW1hem9uYXdzLmNvbVwvdXMtZWFzdC0xX2d4Rk1rZVJ6TiIsInBob25lX251bWJlcl92ZXJpZmllZCI6dHJ1ZSwiY29nbml0bzp1c2VybmFtZSI6InNob3V2aWtUZXN0MiIsImF1ZCI6IjNmcTJrOTJkYmhycnV1YTRpc2oycDZvcXByIiwiZXZlbnRfaWQiOiIwZTUzMzUxMC0zZjQ0LTExZTgtYTBmZi0xNWJlZGJiMWZmMmYiLCJ0b2tlbl91c2UiOiJpZCIsImF1dGhfdGltZSI6MTUyMzY0MjI0NSwicGhvbmVfbnVtYmVyIjoiKzE2MzAzMzU5MjIwIiwiZXhwIjoxNTIzNjQ1ODQ1LCJpYXQiOjE1MjM2NDIyNDUsImVtYWlsIjoic2hvdXZpayt0ZXN0MkBnbWFpbC5jb20ifQ.PdpjAaVBujZVITrte1i9RrvOmmJ1koRxcbeSNWJ_8mfHzBG_SeAtk2ZXG2bPYXAdujIzzNIl3VvMFJA-zPqpSI-_dI0c7hAESJsqwQHDJ5K5FML9DnVaE7h1ik1dmP1q_UkDXkzli3uEbi-Gt_X0rMbfBEZtKFEUnDHIHL7CHdUf0v18jEryKxg0Ro1Of94MZ1_Y2fTfvma67F8L9He_D2ulOAyIJe8V4wkqHeXiB_n-wm8HGkGNze9acPftKuP8_2KfwyOq7v-H-LQOEEtY_7-PKiGu6cTMwY-kHfxswGz-diDCbQ5lvilht3Oo__XNTg3MQvD7SVYz6l9kJOaq5Q"__proto__:
   ObjectRoleSessionName: "web-identity"__proto__: ObjectsessionToken:
   undefinedsts: features.constructor {config: Config, isGlobalEndpoint: true,
   endpoint: Endpoint, _clientId: 6}webIdentityCredentials:
   WebIdentityCredentials {expired: true, expireTime: null, accessKeyId:
   undefined, sessionToken: undefined, params: {…}, …}_clientConfig: {region:
   "us-east-1"}_identityId: nullidentityId: (...)secretAccessKey: undefinedget
   identityId: ƒ ()set identityId: ƒ (identityId)__proto__:
   Credentials__proto__: Object
   - ConsoleLogger.js:84 [DEBUG] 57:25.780 AuthClass - cannot get cognito
   credentials
   - auth.js:140 error signing in:  signin failed

**Comment by richardzcode** [comment]
Is it possible your identity pool doesn't have the user pool setup as one of the providers? In AWS Console, it is in "Authentication Providers" section of the identity pool.

**Comment by shouvik44** [comment]
@richardzcode This was exactly the issue! Thanks for your help figuring this out!

My UserPool and IdentityPools were out of synch. The IdentityPool was pointing to the wrong userpool and the wrong client id. In case anyone runs into this problem, I actually did 2 separate things that caused this.

Here's my use case - which I imagine is fairly common. I am building a mobile app for customers, and a web app for admin users. I want both to use the same AWS services so Admin users can create content the customers will view and stream.

I started out using MobileHub to create both these projects, but made 2 related mistakes. 

1) After creating the mobile app smoothly through MobileHub, I set up the web app using the normal awsmobile init command. This inadvertantly created a new project and therefore Cognito userpool id for my webapp. If you are building a 2nd app in the same project, always use the manual setup so you are using your existing services... not creating new ones.

2) To gain more control over Cognito, I deleted the UserPool -> Client Apps created by MobileHub and then created my own. However, did not go into IdentityPool and change the client apps ids too. If you create your custom client apps through UserPools, make sure you synch your identity pool back to it too.
